staging/src/k8s.io/client-go/tools/auth/exec/exec_test.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package exec
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientauthenticationv1alpha1 "k8s.io/client-go/pkg/apis/clientauthentication/v1alpha1"
+	clientauthenticationv1beta1 "k8s.io/client-go/pkg/apis/clientauthentication/v1beta1"
+	"k8s.io/client-go/rest"
+)
+
+// restInfo holds the rest.Client fields that we care about for test assertions.
+type restInfo struct {
+	host            string
+	tlsClientConfig rest.TLSClientConfig
+	proxyURL        string
+}
+
+func init() {
+	utilruntime.Must(corev1.AddToScheme(scheme))
```

Comment:
why is this needed? modifying the non-test package `scheme` in a test-only file doesn't seem like a great idea... I really want to exercise the methods that use the scheme the way non-test code will

Reply:
I used a `Pod` down below to validate unmarshalling a `runtime.Object` that wasn't a `clientauthentication.ExecCredential`. I'll use a different `runtime.Object` for this so we don't need to modify the `scheme` in the test.

Reply:
962c4f64655